models/dojo.py

```diff
@@ -99,72 +114,328 @@ def add_person(self,arg):
         else:
             print("Error! Individual must be either a fellow or a staff.")
 
-
-    def assign_room(self,arg):
+    def allocate_room(self,arg):
```

Comment:
You seem to be instantiating Dojo multiple times. That is bad practice. A better way to do this is to create a single instance with `dojo = Dojo()` and use `dojo` in place of `Dojo()`